### ISSUE TITLE
OMV7 UI takes almost 10 mins to come up (but can connect with SSH very quickly)

### ISSUE THREAD

**Issue by Tareku99** [openmediavault/openmediavault#1700]
I noticed this right after updating from omv6 to omv7. 

When I had omv6 getting to the UI would take pretty much the same time as it did to connect with SSH.

With OMV7, it takes almost as much as 10 mins to get the UI to work but SSH comes up super fast like usual. 

I tried a fresh install and same issue as well

**Comment by votdev** [comment]
What hardware are we talking about? AMD64 or ARM? Is it then installed via ISO or install script?

have you checked the logs? Any useful information regarding nginx or phpfpm?

**Comment by science-pirate** [comment]
The same problem, something with nginx after upgrading to 7-28. The screenshot shows monitoring alerts after loading the web interface. 
![photo_2024-02-04_16-12-27](https://github.com/openmediavault/openmediavault/assets/86448841/20e5105e-e1e9-4bde-a8be-07f544153a7d)

**Comment by votdev** [comment]
> The same problem, something with nginx after upgrading from 7.0-27 to 7.0-28. The screenshot shows monitoring alerts after loading the web interface.

That is strange because there were no changes for 7.0-27 to 7.0-28.

What is the output of 

```
# journalctl -u nginx.service
```

**Comment by Tareku99** [comment]
` journalctl -u nginx.service
Dec 01 19:30:44 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a rev>
Dec 01 19:30:44 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reve>
Dec 01 23:19:25 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a rev>
Dec 01 23:19:30 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 01 23:19:30 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reve>
Dec 01 23:19:30 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a rev>
Dec 01 23:19:30 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reve>
Dec 01 23:36:57 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a rev>
Dec 01 23:36:57 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 01 23:36:57 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reve>
-- Boot 3c512bfe296e4fdf811b988ebd4a30d5 --
Dec 01 23:37:17 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a rev>
Dec 01 23:37:17 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reve>
Dec 01 23:39:05 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a rev>
Dec 01 23:39:05 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 01 23:39:05 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reve>
-- Boot 7699fef7a49d4421bd88ae07c681847e --
Dec 01 23:39:25 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a rev>
Dec 01 23:39:25 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reve>
Dec 01 23:46:46 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a rev>
Dec 01 23:46:46 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 01 23:46:46 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reve>
-- Boot 6cd4b613e7944939be3aceb819391b32 --
Dec 01 23:49:03 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a rev>
Dec 01 23:49:03 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reve>
Dec 02 12:20:56 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a rev>
Dec 02 12:20:56 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 02 12:20:56 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reve>
Dec 02 12:20:56 openmediavault systemd[1]: nginx.service: Consumed 1.395s CPU time.
-- Boot 734b572a86634fb8b55ca02be116d137 --
Dec 02 12:24:24 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a rev>
Dec 02 12:24:24 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reve>
-- Boot 987f61eb5b684d06ba33f5fa896d06d0 --
Dec 02 12:36:43 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a rev>
lines 1-34...skipping...
Dec 01 19:30:44 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 01 19:30:44 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
Dec 01 23:19:25 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a reverse proxy server...
Dec 01 23:19:30 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 01 23:19:30 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reverse proxy server.
Dec 01 23:19:30 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 01 23:19:30 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
Dec 01 23:36:57 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a reverse proxy server...
Dec 01 23:36:57 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 01 23:36:57 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reverse proxy server.
-- Boot 3c512bfe296e4fdf811b988ebd4a30d5 --
Dec 01 23:37:17 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 01 23:37:17 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
Dec 01 23:39:05 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a reverse proxy server...
Dec 01 23:39:05 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 01 23:39:05 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reverse proxy server.
-- Boot 7699fef7a49d4421bd88ae07c681847e --
Dec 01 23:39:25 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 01 23:39:25 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
Dec 01 23:46:46 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a reverse proxy server...
Dec 01 23:46:46 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 01 23:46:46 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reverse proxy server.
-- Boot 6cd4b613e7944939be3aceb819391b32 --
Dec 01 23:49:03 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 01 23:49:03 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
Dec 02 12:20:56 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a reverse proxy server...
Dec 02 12:20:56 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 02 12:20:56 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reverse proxy server.
Dec 02 12:20:56 openmediavault systemd[1]: nginx.service: Consumed 1.395s CPU time.
-- Boot 734b572a86634fb8b55ca02be116d137 --
Dec 02 12:24:24 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 02 12:24:24 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
-- Boot 987f61eb5b684d06ba33f5fa896d06d0 --
Dec 02 12:36:43 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 02 12:36:43 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
-- Boot a8d29cf492624f5185edb7edbe2e24bb --
Dec 02 13:43:49 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 02 13:43:49 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
-- Boot c6e4678790334674b6c66f55c0fddce8 --
Dec 16 13:16:50 openmediavault systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Dec 16 13:16:50 openmediavault systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
Dec 17 09:41:59 openmediavault systemd[1]: Stopping nginx.service - A high performance web server and a reverse proxy server...
Dec 17 09:41:59 openmediavault systemd[1]: nginx.service: Deactivated successfully.
Dec 17 09:41:59 openmediavault systemd[1]: Stopped nginx.service - A high performance web server and a reverse proxy server.
`

This is the output of journalctl. Nothing strange in the log as far as I can tell. 

Also, just want to make sure, I have been experiencing this error since I upgrade to OMV7, I do not remember the exact version but it might have been around 7.0-24

**Comment by ryecoaaron** [comment]
Where are you seeing this delay and how to recreate?  I have five systems on 7.0-29 and even with a reboot, I'm not seeing any delays on any of them.

**Comment by Tareku99** [comment]
It happens everytime the server comes back on, either from a reboot or a full shutdown. It always has this massive delay for the UI to come up

**Comment by science-pirate** [comment]
> Something useful in `/var/log/nginx/openmediavault-webgui_error.log`?

Nothing, 0B file



> # journalctl -u nginx.service

Nothing in output

**Comment by ryecoaaron** [comment]
@Tareku99 what kind of system? See any errors in the output of: sudo dmesg

**Comment by Tareku99** [comment]
Just wifi related things
```
[    3.388498] iwlwifi 0000:08:00.0: firmware: failed to load iwl-debug-yoyo.bin (-2)
[    3.388506] firmware_class: See https://wiki.debian.org/Firmware for information about missing firmware
[    3.388523] iwlwifi 0000:08:00.0: firmware: failed to load iwl-debug-yoyo.bin (-2)
[  141.712615] iwlwifi 0000:08:00.0: firmware: failed to load iwl-debug-yoyo.bin (-2)
[  141.712664] iwlwifi 0000:08:00.0: firmware: failed to load iwl-debug-yoyo.bin (-2)
```

**Comment by Tareku99** [comment]
Update: After doing this, the wifi keeps crashing every 10 mins or so

**Comment by ryecoaaron** [comment]
What kind of system?  What kernel?  This seems like a kernel issue with that specific wifi device.

**Comment by Tareku99** [comment]
Kernel: Linux 6.1.0-17-amd64
Wifi Card: AX200

**Comment by votdev** [comment]
I'm closing this issue because it seems that OMV is not responsible for that issue. Feel free to discuss the issue here. Maybe other users with this WiFi card  are affected as well.

**Comment by Tareku99** [comment]
How is OMV7 not the cause? I have the server running with no problem for months and right after the update it broke?
I don't get how this is not OMV7 related

**Comment by ryecoaaron** [comment]
Your server updated other things in those months.  The kernel could very easily have a bad driver for your wifi module.  That would explain why it isn't affecting any of my systems.

**Comment by mbgroen** [comment]
> How is OMV7 not the cause? I have the server running with no problem for months and right after the update it broke? I don't get how this is not OMV7 related

Seems more like a Debian 12 issue related to your specific hardware, not OMV specific / related

**Comment by mckaygerhard** [comment]
> I don't get how this is not OMV7 related

https://forum.openmediavault.org/index.php?thread/54800-odd-error-with-webinterface-in-omv-7/&postID=407469#post407469

normally it takes several times to run and show ui in older hardware. seems this will not changed. 

i just link here because i am experience same issue in some deploys.. and not open new issue.. and not found any other solution more than newer disk and more cpu power.. and only happened in the ui web service, OVM5 does not take too much

**Comment by Tareku99** [comment]
A little update for everyone as I still think this is OMV7 specifically since I did not have this issue with the previous version or older updates.

I have moved home, I have a ubiquiti internet setup, I get full gig over wifi and I have ethernet connections in each room, including to the server right now. So I now have full hard wired 1 gig connection in the entire house. 

A speedtest and a speed test within lan was done and I get the speeds I am supposed to. 
Here what was done so far to see what the issue was. I rebooted my server which I would not call old per say (maybe now days it is lol), 64Gb DDR4, ssd for boot drive, i7 6700k and etc. After the reboot, the screen immaterially came up on the terminal, no problem there, logged in and did a speedtest to make sure it was actually connected to the network (got full gig back). 

After I did all of that, I went back to my computer, and I still could not load the webui, after about 3-4 mins, it decided to load. 

Now for the previous convos on here that said it was a driver issue, a kernel issue or any of those. I doubt it is a eth/wifi driver issue considering that there are no issues anywhere else. As far as it being Debian 12 fault or OMW7... OMW7 uses debian 12. OMV6 was perfectly fine and it was not using debian 12. 

I understand that OMV7 doesn't control debian 12 but at the same time it was a choice to MOVE to debian 12, so I expect OMV7 to work better than OMV6. 

Therefore I still think this is OMV7 fault indirectly or directly, whichever one you want to label it as

**Comment by HondaRulez** [comment]
Now I have same issue, runing OMV 7 on PI4. All other system things are very fast (like nfs, disk access, and so on) only the web UI takes 5-10 minutes to load at 1st time...

Loading the page from the localhost (on same Pi) works fine. Opening on any other device (include iOS) very slow..

At the 1st look it seems to be related to the OMV javascript...

**Comment by mckaygerhard** [comment]
yes @HondaRulez  this is confirmed but seems there is not so many solutions .. if you login in using ssh and restart nginx and omv the ui just starts faster.. 

i was trying to get with the root cause but i hate those web craps new technologies..  the OMV uses javascrip cache dependenat things and its hard to debug this wiout get deep into.. there are several reports about same case .. 

but since a server doesn't turn on and off so many times, in my case i just leave this bug as it is because i turn on the server only once and never turn it off again... as workaround i go into ssh, restart nginx and the omv daemon because a server never turns off...
